include/seqan3/alignment/pairwise/alignment_result.hpp

```diff
@@ -19,17 +19,21 @@ namespace seqan3::detail
 {
 
 /*!\brief A struct that contains the actual alignment result data.
- * \tparam id_t          The type for the alignment identifier.
- * \tparam score_t       The type for the resulting score.
- * \tparam back_coord_t  The type for the back coordinate, can be omitted.
- * \tparam front_coord_t The type for the front coordinate, can be omitted.
- * \tparam alignment_t   The type for the alignment, can be omitted.
+ * \tparam id_t                  The type for the alignment identifier.
+ * \tparam score_t               The type for the resulting score.
+ * \tparam back_coord_t          The type for the back coordinate, can be omitted.
+ * \tparam front_coord_t         The type for the front coordinate, can be omitted.
+ * \tparam alignment_t           The type for the alignment, can be omitted.
+ * \tparam score_debug_matrix_t  The type for the score matrix. Only present if seqan3::align_cfg::debug is enabled.
+ * \tparam trace_debug_matrix_t  The type for the trace matrix. Only present if seqan3::align_cfg::debug is enabled.
  */
 template <typename id_t,
           typename score_t,
           typename back_coord_t = std::nullopt_t *,
           typename front_coord_t = std::nullopt_t *,
-          typename alignment_t = std::nullopt_t *>
+          typename alignment_t = std::nullopt_t *,
+          typename score_debug_matrix_t = std::nullopt_t *,
+          typename trace_debug_matrix_t = std::nullopt_t *>
 struct alignment_result_value_type
```

Comment:
Why is the alignment_result type in detail? Aren't people to supposed to know how to access it?

Reply:
That is a internal value type used to store all the members and to selectively disable some of the values in the template argument. This way the user does not need to bother with the template arguments and only sees a simple class that has member functions documented properly. Of course the entire type signature is therefor implementation defined and this is not ideal but for the moment the best compromise I could find.

Reply:
Ok, I don't fully know how it works, I just wanted to bring your attention to it.